packages/core-mobile/app/new/features/buyOnramp/screens/SelectBuyTokenScreen.tsx

```diff
@@ -0,0 +1,187 @@
+import React, { useMemo, useState } from 'react'
+import { SelectTokenScreen } from 'common/screens/SelectTokenScreen'
+import {
+  Icons,
+  Separator,
+  Text,
+  SCREEN_WIDTH,
+  TouchableOpacity,
+  useTheme,
+  View
+} from '@avalabs/k2-alpine'
+import { ListRenderItem } from '@shopify/flash-list'
+import { LocalTokenWithBalance } from 'store/balance'
+import { useSearchableTokenList } from 'common/hooks/useSearchableTokenList'
+import { useErc20ContractTokens } from 'common/hooks/useErc20ContractTokens'
+import { CHAIN_IDS_WITH_INCORRECT_SYMBOL } from 'consts/chainIdsWithIncorrectSymbol'
+import { LogoWithNetwork } from 'features/portfolio/assets/components/LogoWithNetwork'
+import { LoadingState } from 'common/components/LoadingState'
+import { useOnRampToken } from '../store'
+import { useSearchCryptoCurrencies } from '../hooks/useSearchCryptoCurrencies'
+import { isBtcToken, isSupportedToken, isSupportedNativeToken } from '../utils'
+import { CryptoCurrency } from '../types'
+import { MELD_CURRENCY_CODES, ServiceProviderCategories } from '../consts'
+
+type CryptoCurrencyWithBalance = CryptoCurrency & {
+  tokenWithBalance: LocalTokenWithBalance
+}
+
+export const SelectBuyTokenScreen = (): React.JSX.Element => {
+  const {
+    theme: { colors }
+  } = useTheme()
+  const [searchText, setSearchText] = useState<string>('')
+  const [selectedToken, setSelectedToken] = useOnRampToken()
+  const erc20ContractTokens = useErc20ContractTokens()
+  const { filteredTokenList } = useSearchableTokenList({
+    tokens: erc20ContractTokens,
+    hideZeroBalance: false
+  })
+  const { data: cryptoCurrencies, isLoading: isLoadingCryptoCurrencies } =
+    useSearchCryptoCurrencies({
+      categories: [ServiceProviderCategories.CryptoOnramp]
+    })
+
+  const handleSelectToken = (token: CryptoCurrency): void => {
+    setSelectedToken(token)
+    // @ts-ignore TODO: make routes typesafe
+    // navigate('./inputAmount')
+  }
+
+  const supportedCryptoCurrencies = useMemo(() => {
+    return cryptoCurrencies?.reduce((acc, crypto) => {
+      const token = filteredTokenList.find(
+        tk =>
+          isSupportedNativeToken(crypto, tk) ||
+          isSupportedToken(crypto, tk) ||
+          isBtcToken(crypto, tk)
```

Comment:
maybe we can have a util for this one? `isTokenSupportedForBuying` or `isSupportedForBuying` ?